tests/dash/gnmi_utils.py

```diff
@@ -0,0 +1,322 @@
+import logging
+import json
+import time
+import re
+import ipaddress
+import pytest
+import socket
+import uuid
+
+from dash_api.appliance_pb2 import *
+from dash_api.vnet_pb2 import *
+from dash_api.eni_pb2 import *
+from dash_api.qos_pb2 import *
+from dash_api.route_pb2 import *
+from dash_api.route_rule_pb2 import *
+from dash_api.vnet_mapping_pb2 import *
+from dash_api.route_type_pb2 import *
+from dash_api.types_pb2 import *
+
+logger = logging.getLogger(__name__)
+
+GNMI_PORT = 0
+# Wait 15 seconds after starting GNMI server
+GNMI_SERVER_START_WAIT_TIME = 30
+GNMI_CONTAINER = "gnmi"
+GNMI_PROGRAM = "gnmi-native"
+GNMI_CA_CERT = "gnmiCA.pem"
+GNMI_CA_KEY = "gnmiCA.key"
+GNMI_SERVER_CERT = "gnmiserver.crt"
+GNMI_SERVER_KEY = "gnmiserver.key"
+GNMI_CLIENT_CERT = "gnmiclient.crt"
+GNMI_CLIENT_KEY = "gnmiclient.key"
+
+
+# ENABLE_PROTO = False
+ENABLE_PROTO = True
+
+
+def gnmi_port(duthost):
```

Comment:
>gnmi_port [](http://example.com/codeflow?start=4&length=9)

For each utils function, add some function level comments to explain the functionality, input, output.

Reply:
updated